src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -115,4 +123,80 @@ public static void addNewUser(String firstName, String lastName, String username
       logger.log(Level.SEVERE, exception.getMessage(), exception);
     }
   }
+
+  /**
+   * Queries the mails of users to notify and returns them in a single string. Includes url, user
+   * and password to give the user the ability to choose between local and cloud SQL variables.
+   */
+  private String getUserEmailsAsString(List<Integer> userIds, String SQL_URL, String SQL_USER, 
+                                       String SQL_PASSWORD) {
+    String userEmails = new String();
+    for (int userId : userIds) {
+      // Query the email of the current user.
+      String query = "SELECT email FROM User WHERE id = " + userId;
+      try (Connection connection = DriverManager.getConnection(SQL_URL, SQL_USER, SQL_PASSWORD);
+        PreparedStatement pst = connection.prepareStatement(query);
+        ResultSet rs = pst.executeQuery()) {
+        rs.next();
+        // Concatenate the user's email and a comma for the InternetAddress parser to separate.
+        userEmails = userEmails.concat(rs.getString(1));
+        userEmails = userEmails.concat(",");
+        connection.close();
+      } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(EmailServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    }
+    // Erase the last comma.
+    userEmails = userEmails.substring(0, userEmails.length() - 1);
+    return userEmails;
+  }
+
+  /**
+   * Queries IDs of the author of the modified question/answer and its followers. Includes url, 
+   * user and password to give the user the ability to choose between local and cloud SQL 
+   * variables.
+   */
+  private List<Integer> getUsersToNotify(String typeOfNotification, int modifiedElementId,
+                                         String SQL_URL, String SQL_USER, String SQL_PASSWORD) {
+    List<Integer> usersToNotify = new ArrayList<>();
+    if(typeOfNotification.equals("question")) {
```

Comment:
nit: Space after 'if'

```suggestion
    if (typeOfNotification.equals("question")) {
```

Reply:
Done!